pkg/daemon/ceph/osd/daemon.go

```diff
@@ -196,6 +198,27 @@ func Provision(context *clusterd.Context, agent *OsdAgent, crushLocation string)
 			// set it's a metadata device
 			rawMetadataDevice.Type = pvcMetadataTypeDevice
 			rawDevices = append(rawDevices, rawMetadataDevice)
+		// We have a metadata device and wal device
+		case 3:
+			rawMetadataDevice, err := clusterd.PopulateDeviceInfo(agent.devices[1].Name, context.Executor)
```

Comment:
Several of these lines are duplicated from case 2. How about refactoring to a helper method?